fdbclient/FDBTypes.h

```diff
@@ -1329,6 +1330,21 @@ struct TenantMode {
 
 	uint32_t mode;
 };
+struct GRVCacheSpace {
+	Version cachedReadVersion;
+	double lastGrvTime;
+
+	GRVCacheSpace() : cachedReadVersion(Version(0)), lastGrvTime(0.0) {}
+};
+
+// This structure can be extended in the future to include additional features that required a shared state
+struct DatabaseSharedState {
+	Mutex mutexLock;
```

Comment:
Is there any way we can introduce some sort of protocol version notion into this object? Since we are sharing it between client libraries, it may be a useful sanity check that our sharing is always within a single version where this object hasn't changed.

Reply:
I added a `Future<ProtocolVersion>` field which gets set in `DatabaseContext::initSharedState`. Does that align with the type of usage you'd expect?

Reply:
I think this could be set without a future at object construction time based on the protocol version of the client library that creates it and could be made const. Once we have it, we would want the operation that sets the shared state to confirm (ASSERT?) that the protocol version stored in the state object matches the protocol version of the current object. The last thing is that I would put this as the first field in your struct to avoid some types of problems reading it if the object format changes.